### ISSUE TITLE
Ошибка создания конфига при установке через chocolatey

### ISSUE THREAD

**Issue by viktor02** [BladeMight/Mahou#280]
## Problem/Request description

При установке через chocolatey программа не может создать файлы. Нажатие Да или Нет ничего не решают, при следующем запуске снова появляется это окно.

## Expected behavior (Only if reporting bug)

Конфиги по умолчанию в аппдате или папке пользователя? 

## System information

- Windows version: Microsoft Windows [Version 10.0.19041.329]
- Windows architecture: x64
- .Net version: 4.8 (528372)
- Mahou version: latest from chocolatey

## Image
![изображение](https://user-images.githubusercontent.com/19672960/88835664-40e64880-d1de-11ea-9835-8bc25a3cbae7.png)
![изображение](https://user-images.githubusercontent.com/19672960/88836532-7f303780-d1df-11ea-9491-18556ca9b933.png)

**Comment by viktor02** [comment]
Проблема, конечно, решается через "Запуск от администратора", но все же.

**Comment by BladeMight** [comment]
@viktor02

> Конфиги по умолчанию в аппдате или папке пользователя?

По умолчанию - "В папке где Mahou.exe", при нажатии на ОК в [попробовать](https://user-images.githubusercontent.com/19672960/88835664-40e64880-d1de-11ea-9835-8bc25a3cbae7.png), Mahou попробует всё же записать в папку где Mahou.exe значение что "будет использоваться настройка в AppData", но также если не получится создаст в `%AppData%\Mahou` файл `.force` который будет указывать что Mahou будет использовать настройки в AppData принудительно. Проверь есть ли такой в `%AppData%\Mahou` у тебя.

Однако же версия Mahou в Chocolatey довольно старая, я рекомендую использовать latest-commit. (нужно наверно будет и её тоже добавить в Chocolatey....)


Для обновления до latest-commit:
1. Во вкладке "Обновления" поставить канал обновлений на LatestCommit(Beta), и обновиться как обычно(проверить, скачать обновление)
2. На странице: https://github.com/BladeMight/Mahou/releases/tag/latest-commit есть постоянно авто-обновляемые сборки. (если не знаешь какую брать - Release_x86_x64.zip)

**Comment by viktor02** [comment]
> Проверь есть ли такой в %AppData%\Mahou у тебя.

Да, есть. Но, как я уже и сказал, я решил эту проблему запуском от админа и выставлением прав на папку в programdata

Попытался обновить через программу, скачалось, но "папка tools используется, закройте все программы", хотя в диспетчере задач процесса mahou я не увидел. Но судя по постфиксу -dev она вроде обновилась. Для верности обновил exe-шник вручную)

**Comment by BladeMight** [comment]
> Но судя по постфиксу -dev она вроде обновилась. Для верности обновил exe-шник вручную)

Да, и так же там в `<>` последний коммит по которому собран Mahou.

**Comment by viktor02** [comment]
>И всё равно выдаёт "попробовать" при запуске?

Уже нет. Но я не знаю почему в течении двух попыток запуска без админ-прав мне выдавало это сообщение. После одного запуска с админ правами следующие запуски без админ-прав спокойно работают без всяких сообщений. Уж извините за тавтологию)

**Comment by BladeMight** [comment]
>  После одного запуска с админ правами

Рядом с Mahou.exe появился Mahou.ini? А в нём `AppDataConfigs=True`, если да, тогда он просто теперь не пытается его "записать", только читает значение `AppDataConfigs`, и меняет папку на `%AppData%\Mahou` где далее делает и запись и чтение настроек, сниппетов и т.д.

**Comment by viktor02** [comment]
![изображение](https://user-images.githubusercontent.com/19672960/88860787-2ffcfd80-d205-11ea-92df-14c20a34dcc6.png)
Рядом с mahou.exe - нет.
![изображение](https://user-images.githubusercontent.com/19672960/88860809-43a86400-d205-11ea-9db8-d7dffdef8c7b.png)
Сохраняется в appdata
![изображение](https://user-images.githubusercontent.com/19672960/88860875-6470b980-d205-11ea-9060-b2bbe5fb807f.png)

**Comment by BladeMight** [comment]
Значит всё таки работает из-за `.force`, как я и говорил раннее.

**Comment by viktor02** [comment]
В смысле exe-шник? А зачем? Настройки вроде бы все равно сохраняются, а возможность обновлять через choco, как я понимаю, пропадет.

**Comment by BladeMight** [comment]
@viktor02 Посмотрю и подумаю что можно сделать с "пре-релизами chocolatey"...

**Comment by viktor02** [comment]
Кстати, попробовал сейчас снова(переустанавливал винду), проблема та же :) 
Версия в chocolatey Mahou 2.9.0.1

**Comment by viktor02** [comment]
Обновил на последний коммит, от 2020-11-17, всё так же. Видимо создать папку не может без админ-прав. Соответственно в appdata и в папке chocolatey конфигов нет. 
Решение только запустить из под админа один раз, чтобы создал папки, дальше можно запускать от прав пользователя.
